core-api/src/main/java/org/silverpeas/core/personalization/UserPreferences.java

```diff
@@ -62,18 +66,19 @@
   public UserPreferences() {
   }
 
-  public UserPreferences(String userId, String language, String look,
-      String collaborativeWorkSpaceId, boolean thesaurusEnabled,
-      boolean dragAndDropEnabled, boolean webdavEditionEnabled, UserMenuDisplay display) {
-    this(language, look, collaborativeWorkSpaceId, thesaurusEnabled, dragAndDropEnabled,
+  public UserPreferences(String userId, String language, final ZoneId zoneId, String look,
+      String collaborativeWorkSpaceId, boolean thesaurusEnabled, boolean dragAndDropEnabled,
+      boolean webdavEditionEnabled, UserMenuDisplay display) {
+    this(language, zoneId, look, collaborativeWorkSpaceId, thesaurusEnabled, dragAndDropEnabled,
         webdavEditionEnabled, display);
     setId(userId);
   }
 
-  public UserPreferences(String language, String look, String collaborativeWorkSpaceId,
-      boolean thesaurusEnabled, boolean dragAndDropEnabled, boolean webdavEditionEnabled,
-      UserMenuDisplay display) {
+  public UserPreferences(String language, final ZoneId zoneId, String look,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00107)